### PR TITLE
Feature/offline mode 66

### PR DESCRIPTION
**Pull request by helkhour**
This PR introduces foundational offline support to the app. Initial work includes network monitoring logic and basic connectivity-based navigation updates.

Key Changes:

- Network Connectivity Observers:
    Added ConnectivityObserver and NetworkConnectivityObserver to track real-time network status changes and respond accordingly.

- MainActivity Updates:
    Integrated network observers to handle connectivity changes at the app's entry point. Currently, it displays an "Unavailable Network" message. Once #67 is complete, this should lead the user to a dedicated offline experience.

- Offline Screens:
    Added preliminary drafts of OfflinePracticeQuestionsScreen and OfflineScreen as previews. These screens are still in early stages and will be refined in future updates.

- Build and Dependencies:
    Updated build.gradle.kts and libs.versions.toml to support new features and enhance compatibility.

- Tests:
    Added tests for the network observer to verify accurate detection and handling of connectivity changes.
   
Do you think I should focus on anything else ? Currently, this offline mode addresses only random network disconnections. It does not yet support a full [offline-first approach](https://developer.android.com/topic/architecture/data-layer/offline-first), but can absolutely work on it if you think it is a good idea !

### CODE REVIEW

**Comment by helkhour** [comment]
NB : I forgot to run ktfmtFormat. 
I will try to do it tomorrow in the morning. Do not merge my request in the meantime

File: app/src/main/java/com/github/se/orator/ui/navigation/NavigationActions.kt [hunk 1]
```diff
@@ -51,6 +53,9 @@ object Screen {
   const val FEEDBACK = "Feedback Screen"
   const val CHAT_SCREEN = "chat_screen"
   const val SPEAKING = "Speaking"
+  const val PRACTICE_QUESTIONS_SCREEN = "PracticeQuestions Screen"
```
Comment (MathisHage) [inline app/src/main/java/com/github/se/orator/ui/navigation/NavigationActions.kt:56]: Is there a difference between `PRACTICE_QUESTIONS_SCREEN` and the unused constant `PRACTICE_SCREEN` ?

File: app/src/main/java/com/github/se/orator/ui/network/OfflinePracticeQuestionsScreen.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        // Display each question in the list with a clickable action
+        questions.forEach { question ->
+            Text(
+                text = question,
+                fontSize = 16.sp,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(vertical = 8.dp)
+                    .clickable {
+                        // Here we need to navigate to the recording screen and pass the selected question as a parameter
```
Comment (MathisHage) [inline app/src/main/java/com/github/se/orator/ui/network/OfflinePracticeQuestionsScreen.kt:59]: Two options are possible : 
* creating a new `OfflineRecordingViewModel.kt`file to simply record audio and store the file, or
* Adapting the files `SpeakingViewModel.kt` and `SpeakingRepository.kt` to have an online mode and an offline mode.

This is outside of the scope of this PR though, this comment is just to respond to the comments on the file.

**Review by MathisHage — CHANGES_REQUESTED** [review]
Nicely done ! Everything is clear and clean. Need to run KTFMT though :smiley_cat: !

File: app/src/main/java/com/github/se/orator/ui/navigation/NavigationActions.kt (hunk 1 above)
Comment (helkhour) [reply app/src/main/java/com/github/se/orator/ui/navigation/NavigationActions.kt:56]: oops- missed that

**Comment by helkhour** [comment]
Sonar Cloud is not passing cause I forgot to push it. Will link the code in #67
